department-of-veterans-affairs/caseflow
# Show ihp details on the front end

Implements the front end of #15405

### Description
Adds a tool tip to queues to display information on ihp tasks associated with appeals

### Acceptance Criteria
- [x] Judge Assign queue shows the IHP timing details on hover
- [x] Atty Assigned queue shows the IHP timing details on hover
- [x] Attorney On Hold queue shows the IHP timing details on hover

### Testing Plan
1. Sign in as bvaaaabshire and click into a legacy appeal from your queue
1. Create an admin action for ihp
1. Complete the task as the colocated user
1. Reload the judge's queue
1. Hover over days waiting for the same case and confirm ihp was requested and received
- [x] Judge Assign queue shows the IHP timing details on hover for legacy
6. Repeat the same for an ama appeal
- [x] Judge Assign queue shows the IHP timing details on hover for ama
7. Sign in as bvacasper1 and select a legacy appeal from you assigned queue
1. Create an admin action for ihp
1. Confirm ihp request information shows up in your on hold tab
- [x] Attorney On Hold queue shows the IHP timing details on hover for legacy
10. Repeat the same for an ama appeal
- [x] Attorney On Hold queue shows the IHP timing details on hover for ama
11. Select a legacy appeal from your assigned queue and make some ihp tasks for it
```ruby
vacols_id = "2226048"
appeal = LegacyAppeal.find_by(vacols_id: vacols_id)
InformalHearingPresentationTask.create(
  assigned_to: Vso.first,
  created_at: 60.days.ago,
  appeal: appeal
).completed!
```
12. Confirm ihp request information shows up in your on assigned tab on reload
- [x] Atty Assigned queue shows the IHP timing details on hover for legacy
13. Repeat the same for ama
- [x] Atty Assigned queue shows the IHP timing details on hover for ama

### User Facing Changes

 BEFORE|AFTER
 ---|---
none|<img width="1082" alt="Screen Shot 2020-11-10 at 12 44 01 PM" src="https://user-images.githubusercontent.com/45575454/98711174-abb52d00-2352-11eb-80e7-339d26b475c4.png">



### Code Documentation Updates

### Storybook Story

1. Start up your storybook server
```bash
make storybook
```
2. Navigate to http://localhost:6006/?path=/story/queue-components-ihpdayswaitingtooltip
3. Play with the new component! (still have not found a great way to change the dates nicely)
